### ISSUE TITLE
Using the "except" option with "first-nested" does not work for "custom-property-empty-line-before" when using `postcss-styled-syntax`

### ISSUE THREAD

**Issue by insidewhy** [stylelint/stylelint#6766]
### What steps are needed to reproduce the bug?

When using `["always", { "except": ["first-nested"] }]` as the option for `custom-property-empty-line-before`, `stylelint` is still demanding an empty line before the very first css variable defined in a rule:

```typescript
const Form = styled.form`
  --max-width: 58rem;
`
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard"
  ],
  "customSyntax": "postcss-styled-syntax",
  "rules": {
    "custom-property-empty-line-before": ["always", {
      "except": ["first-nested"]
    }]
  }
}
```

### How did you run Stylelint?

`yarn stylelint '**/*.{ts,tsx}'`

### Which version of Stylelint are you using?

`15.4.0`

### What did you expect to happen?

I should not have to add a newline before the variable definition, prettier actually automatically removes this line which makes it impossible to add the newline. 

### What actually happened?

The only way for me to work around it is to add a `/* stylelint-disable-next-line */` as the first line since prettier would always reformat the code to remove the line.

### Does the bug relate to non-standard syntax?

I'm using `postcss-styled-syntax` and I don't know if the bug manifests without it

### Proposal to fix the bug

Make the configuration work as documented.

**Comment by mattxwang** [comment]
Hi @insidewhy, thanks for submitting an issue and for following the template.

It's a bit tricky for us to reproduce this error since it seems like it depends on the rest of your project setup. Would you be able to create a minimum reproducible example, e.g. a repository with *just* the relevant packages / JS files to cause the error you've described? For example, what version of styled components you're using, etc.

**Comment by ybiquitous** [comment]
@insidewhy Thanks for the report. Probably, [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBc4IDsDOAXABAMQgJwFtMBeTDATwBsYoA6AM3wIAMAdZTTAWi4IEMAHlwDuASyjoAFvEwBWABx4YBANztma5CAA0IeqOoA5PgTiIYAkwAdqtAFaoBOpMn0BzBCGDtOrEBfQYZChUPxkAbR9OTD8KakpRZHQuSFdRNy4MPmC+PCg-KIBdbSi-MABXDAgCAGVyJMEwmJArCAwwVFRM9CoaTPr0Rp1SkDxy6lCETG8OaLLK9GquKzwIKxg8Hq5lKy2E5BguACMYRiUmyJA+SmE+ckntaajo5oswGF2Lv308DC4DjA0PyFZ6YAC+INmYPYYOcqXcuEIfHQngcKGcsCsqE8M18LTa6A6XTifVQAyGMj8lGRMAwfhKs1iPXiiXQTSpNLpw0ZIBJ+2S8PS3WyUFy+SmHMCXIZeJ661QYDwok+EpA1KlbJAMJAYKAA) can produce the problem.

<img width="726" alt="Screenshot 2023-04-09 at 23 04 08" src="https://user-images.githubusercontent.com/473530/230777356-807afb93-ecf0-45e9-8c73-18ff29635c3f.png">

---

Also, as @mattxwang commented, this problem relates to `postcss-styled-syntax`. Our `isFirstNested()` util checks a parent node type of a `Declaration` node as follows:

https://github.com/stylelint/stylelint/blob/9a71709607bf58e623717301707caa3920bc00f2/lib/rules/custom-property-empty-line-before/index.js#L91

https://github.com/stylelint/stylelint/blob/9a71709607bf58e623717301707caa3920bc00f2/lib/utils/isFirstNested.js#L12-L14

`postcss-styled-syntax` procuces a `Root` node instead of a `Rule` node (standard CSS). See the example:

`example.js`:

```js
const css = `
a {
  --max-width: 58rem;
}
`;

require('postcss')
	.parse(css)
	.walkDecls((decl) => {
		console.log(decl);
	});

const jsx = `
const Form = styled.form\`
  --max-width: 58rem;
\`;
`;

require('postcss-styled-syntax')
	.parse(jsx)
	.walkDecls((decl) => {
		console.log(decl);
	});
```

Run:

```console
$ node example.js
<ref *1> Declaration {
  raws: { before: '\n  ', between: ': ' },
  type: 'decl',
  parent: Rule {
    raws: { before: '\n', between: ' ', semicolon: true, after: '\n' },
    type: 'rule',
    nodes: [ [Circular *1] ],
    parent: Root {
      raws: [Object],
      type: 'root',
      nodes: [Array],
      source: [Object],
      lastEach: 1,
      indexes: [Object],
      [Symbol(isClean)]: false,
      [Symbol(my)]: true
    },
    source: { start: [Object], input: [Input], end: [Object] },
    selector: 'a',
    lastEach: 1,
    indexes: { '1': 0 },
    [Symbol(isClean)]: false,
    [Symbol(my)]: true
  },
  source: {
    start: { offset: 7, line: 3, column: 3 },
    input: Input {
      css: '\na {\n  --max-width: 58rem;\n}\n',
      hasBOM: false,
      id: '<input css 1sZ71K>',
      [Symbol(fromOffsetCache)]: [Array]
    },
    end: { offset: 25, line: 3, column: 21 }
  },
  prop: '--max-width',
  value: '58rem',
  [Symbol(isClean)]: false,
  [Symbol(my)]: true
}
<ref *1> Declaration {
  raws: { before: '\n  ', between: ': ' },
  type: 'decl',
  parent: Root {
    raws: {
      styledSyntaxRangeStart: 26,
      styledSyntaxRangeEnd: 49,
      isRuleLike: true,
      styledSyntaxIsComponent: true,
      semicolon: true,
      after: '\n',
      codeBefore: '\nconst Form = styled.form`',
      codeAfter: '`;\n'
    },
    type: 'root',
    nodes: [ [Circular *1] ],
    source: { input: [Input], start: [Object] },
    parent: Document {
      raws: {},
      type: 'document',
      source: [Object],
      nodes: [Array],
      lastEach: 1,
      indexes: [Object],
      [Symbol(isClean)]: false,
      [Symbol(my)]: true
    },
    lastEach: 1,
    indexes: { '1': 0 },
    [Symbol(isClean)]: false,
    [Symbol(my)]: true
  },
  source: {
    start: { offset: 29, line: 3, column: 3 },
    input: Input {
      css: '\n  --max-width: 58rem;\n',
      hasBOM: false,
      id: '<input css Rfl0_Q>',
      [Symbol(fromOffsetCache)]: [Array]
    },
    end: { offset: 47, line: 3, column: 21 }
  },
  prop: '--max-width',
  value: '58rem',
  [Symbol(isClean)]: false,
  [Symbol(my)]: true
}
```
